ignite/engine/deterministic.py

```diff
@@ -179,19 +179,18 @@ class DeterministicEngine(Engine):
     def __init__(self, process_function: Callable[[Engine, Any], Any]):
         super(DeterministicEngine, self).__init__(process_function)
         self.state_dict_user_keys.append("rng_states")
+        if not hasattr(self.state, "rng_states"):
+            setattr(self.state, "rng_states", None)
         self.add_event_handler(Events.STARTED, self._init_run)
         self.add_event_handler(Events.DATALOADER_STOP_ITERATION | Events.TERMINATE_SINGLE_EPOCH, self._setup_seed)
 
     def state_dict(self) -> OrderedDict:
+        setattr(self.state, "rng_states", _get_rng_states())
         state_dict = super(DeterministicEngine, self).state_dict()
-        state_dict["rng_states"] = _get_rng_states()
```

Comment:
Let's keep this, instead of `setattr(self.state, "rng_states", _get_rng_states())`